usr/src/uts/common/fs/zfs/zap_micro.c

```diff
@@ -185,15 +187,30 @@ zap_name_alloc(zap_t *zap, const char *key, matchtype_t mt)
 	zn->zn_key_orig = key;
 	zn->zn_key_orig_numints = strlen(zn->zn_key_orig) + 1;
 	zn->zn_matchtype = mt;
+	zn->zn_normflags = zap->zap_normflags;
```

Comment:
I think we could move this after `zn->zn_hash = zap_hash(zn);` for readability purposes.